File: benchmarks/auth-db-http-server/fastify/src/routes/raw-route.mjs
export function registerRawRoute({
  fastify,
  auth,
  baseUrl,
  authenticate,
}) {
  fastify.get('/bench/raw', async (request, reply) => {
    const email = await authenticate({
      auth,
      baseUrl,
      nodeHeaders: request.headers,
    });
    if (email === null) {
      reply.code(401).type('application/json; charset=utf-8');
      return JSON.stringify({ error: 'unauthorized' });
    }

    reply.type('application/json; charset=utf-8');
    return JSON.stringify({ email, value: 'raw benchmark value' });
  });
}

File: benchmarks/auth-db-http-server/dart_edge_http_server/lib/src/routes/raw_route.dart
import 'dart:convert';
import 'dart:io';

import 'package:dart_edge_auth/dart_edge_auth.dart';
import 'package:dart_edge_http_server/dart_edge_http_server.dart';

import '../services.dart';

final class RawRoute extends HttpRouteDefinition<Services, Object?> {
  @override
  RouteOptions get options => RouteOptions(
    operationId: 'benchmarkRaw',
    success: ResponseSpec.json(),
    errors: [ErrorResponse.unauthorized(code: 'unauthorized')],
  );

  @override
  Future<Object?> handle(RequestContext<Services> ctx) async {
    final email = ctx.requireAuthIdentity.email;
    if (email == null) {
      return ctx.res.code(HttpStatus.unauthorized).json({
        'error': 'unauthorized',
      });
    }

    return ctx.res
        .type('application/json; charset=utf-8')
        .send(jsonEncode({'email': email, 'value': 'raw benchmark value'}));
  }
}
